tests/testlib/s2n_key_schedule_testlib.c

```diff
@@ -59,3 +58,18 @@ S2N_RESULT s2n_connection_set_test_master_secret(struct s2n_connection *conn,
     conn->secrets.extract_secret_type = S2N_MASTER_SECRET;
     return S2N_RESULT_OK;
 }
+
+S2N_RESULT s2n_connection_set_test_message_type(struct s2n_connection *conn, message_type_t expected_message_type)
+{
+    /* scan over all handshake types until the desired message type is found */
+    for (uint32_t handshake = 0; handshake < S2N_HANDSHAKES_COUNT; handshake++) {
+        for (int message = 0; message < S2N_MAX_HANDSHAKE_LENGTH; message++) {
```

Comment:
This doesn't seem right, but I could be missing something. Before, it seems that handshake_type was set in advance, and the message number was retrieved for the given handshake type and message type. Now, it seems that the first message number of a message type is given for any handshake type. Am I understanding this correctly?

Reply:
You are understanding it correctly, but the tests are independent of the `handshake_type` and only care about the message type.

Also note that the set handshake type was not actually the used one. The state machine is a two dimensional array of 256 rows x 32 columns. The previous test code was indexing into the state machine like this
```
state_machine[4][36]
# 4 is a valid row

# 36 is not a valid column, and is actually yielding an 
# element from the next row, which is handshake type 5
```

I added a comment explaining this behavior in more detail. 

If we want to prevent customers from relying on this behavior, we could make the function random. If there are `n` (`handshake_type`, `message_number`) combinations that yield the desired `message_type`, we could randomly choose 1 of those `n` combinations, so that that any test relying on a specific `handshake_type` would be flaky. That seemed unnecessary but let me know if you have a different perspective.